tests/core/engine_adapter/test_bigquery.py

```diff
@@ -670,6 +670,106 @@ def test_comments(make_mocked_engine_adapter: t.Callable, mocker: MockerFixture)
     ]
 
 
+def test_nested_comments(make_mocked_engine_adapter: t.Callable, mocker: MockerFixture):
+    adapter = make_mocked_engine_adapter(BigQueryEngineAdapter)
+    allowed_column_comment_length = BigQueryEngineAdapter.MAX_COLUMN_COMMENT_LENGTH
+
+    execute_mock = mocker.patch(
+        "sqlmesh.core.engine_adapter.bigquery.BigQueryEngineAdapter.execute"
+    )
+
+    nested_columns_to_types = {
+        "record": exp.DataType.build(
+            "STRUCT<'int_field': INT, 'record_field': STRUCT<'sub_record_field': STRUCT<'nest_array' ARRAY<INT64>>>"
+        ),
+        "repeated_record": exp.DataType.build(
+            "ARRAY<STRUCT<'nested_repeated_record' ARRAY<STRUCT<'int_field': INT, 'array_field': ARRAY<INT64>, 'struct_field' STRUCT<'nested_field': INT>>>>>"
+        ),
+    }
+
+    long_column_descriptions = {
+        "record": "1" * (allowed_column_comment_length * 2),
+        "record.int_field": "2" * (allowed_column_comment_length * 2),
+        "record.record_field": "3" * (allowed_column_comment_length * 2),
+        "record.record_field.sub_record_field": "4" * (allowed_column_comment_length * 2),
+        "record.record_field.sub_record_field.nest_array": "5"
+        * (allowed_column_comment_length * 2),
```

Comment:
For instance this bit is awkward to read. Not sure what the multiplication offers in this test.